src/test/scala/com/gu/support/workers/lambdas/CreatePaymentMethodStateDecoderSpec.scala

```diff
@@ -46,31 +47,27 @@ class CreatePaymentMethodStateDecoderSpec extends FlatSpec with Matchers with Mo
   }
 
   "CreatePaymentMethodStateDecoder" should "be able to decode a contribution with PayPal payment fields" in {
-    val state = decode[CreatePaymentMethodState](createPayPalPaymentMethodContributionJson())
-    state.isRight should be(true)
-    val result = state.right.get
-    result.product match {
-      case contribution: Contribution => contribution.amount should be(5)
-      case _ => fail()
-    }
-    result.paymentFields match {
-      case paypal: PayPalPaymentFields => paypal.baid should be(validBaid)
-      case _ => fail()
-    }
+    val maybeState = decode[CreatePaymentMethodState](createPayPalPaymentMethodContributionJson())
+
+    val fieldsToTest = maybeState.map(state =>
+      (state.accessScope, state.product, state.paymentFields))
+    fieldsToTest should be(Right(
+      AccessScopeNoRestriction,
+      Contribution(5, GBP, Monthly),
+      PayPalPaymentFields(validBaid)
+    ))
```

Comment:
I changed the tests to assert everything in one go so that we don't get failures just saying "true was not false" or vice versa